core/src/main/scala/cats/data/Chain.scala

```diff
@@ -574,6 +574,62 @@ sealed abstract class Chain[+A] {
    */
   final def size: Long = length
 
+  /**
+   * The number of elements in this chain, if it can be cheaply computed, -1 otherwise.
+   * Cheaply usually means: Not requiring a collection traversal.
+   */
+  final def knownSize: Long =
+    this match {
+      case _ if isEmpty       => 0
+      case Chain.Singleton(_) => 1
+      case _                  => -1
```

Comment:
could we add `case Chain.Wrap(seq) => seq.knownSize` here?

Reply:
Yeah, I tried it in the first place, but turned out that 2.12 does not have `knownSize` implemented. So if we really need it then we'll have to deal with the cross-build compatibility here somehow. Not sure what is the best way to approach that. Any ideas?

Reply:
I mean, I can think about two approaches:
1. Implement a compat version of `knownSize` for `Seq` for 2.12.
2. Handle `case Wrap(seq)` in 2.13+ only.